monai/apps/auto3dseg/auto_runner.py

```diff
@@ -251,84 +272,42 @@ def check_cache(self):
             working directory, the result will be ``empty_cache`` in which all ``has_cache`` keys are
             set to False.
         """
-        empty_cache = {
-            "analyze": {"has_cache": False, "datastats": None},
-            "algo_gen": {"has_cache": False},
-            "train": {"has_cache": False},
-        }
+
+        empty_cache = {"analyze": False, "datastats": None, "algo_gen": False, "train": False}
+
         if self.not_use_cache or not os.path.isfile(self.cache_filename):
             return empty_cache
 
         cache = ConfigParser.load_config_file(self.cache_filename)
 
-        if cache["analyze"]["has_cache"]:
-            # check if the file in the right format and exists.
-            if not isinstance(cache["analyze"]["datastats"], str):
-                cache["analyze"] = False
-                cache["analyze"]["datastats"] = None
+        for k, v in empty_cache.items():
+            cache.setdefault(k, v)
 
-            if not os.path.isfile(cache["analyze"]["datastats"]):
-                cache["analyze"]["has_cache"] = False
+        if cache["analyze"]:
+            if not (isinstance(cache["datastats"], str) and os.path.isfile(cache["datastats"])):
+                cache["analyze"] = False
+                cache["datastats"] = None
 
-        if cache["algo_gen"]["has_cache"]:
+        if cache["algo_gen"]:
             history = import_bundle_algo_history(self.work_dir, only_trained=False)
             if len(history) == 0:  # no saved algo_objects
-                cache["algo_gen"]["has_cache"] = False
+                cache["algo_gen"] = False
 
-        if cache["train"]["has_cache"]:
+        if cache["train"]:
             trained_history = import_bundle_algo_history(self.work_dir, only_trained=True)
             if len(trained_history) == 0:
-                cache["train"]["has_cache"] = False
+                cache["train"] = False
 
         return cache
 
-    def export_cache(self):
+    def export_cache(self, **kwargs):
         """
         Save the cache state as ``cache.yaml`` in the working directory
         """
-        ConfigParser.export_config_file(self.cache, self.cache_filename, fmt="yaml", default_flow_style=None)
-
-    def check_analyze(self, analyze: bool):
```

Comment:
I think it's nice to simplify these checks, perhaps we can remove some of the `set_*` method as well? cc @mingxin-zheng

Reply:
I can certainly remove some of them, and there are many   (set_num_fold, set_training_params, set_prediction_params, set_analyze_params, set_hpo_params, set_nni_search_space, set_image_save_transform)

but I'm not sure what's better, when having those methods, a user will see them (which is an indicator what he can change something). 
For example, right now, to set number of folds
`runner.set_num_fold(n_fold = 2)`
but instead we simply update a variable directly
`runner.num_fold = 2`
to accomplish the same.  But without the setters method, it might be difficult for a user to see what's possible to change

we could merge this PR as is, and then continue further improvements with next PRs

Reply:
sure, please feel free to update or ignore this, perhaps @mingxin-zheng can decide as he is the original designer.

Reply:
Hi @wyli , my thoughts were the similar as @myron 's in the beginning: the user `AutoRunner` was supposed to be beginners in medical segmentation tasks who may need a clearly documented function. 

Even so, if the usage scenario is preferred to be `runner.num_fold = 2`, it will be a breaking change and I agree that can be another MR.